File: LeetCode/container_with_most_water.py
#!/usr/bin/env python
#coding: utf-8

class Solution:
    # @return an integer
    def maxArea(self, height):
        low, high = 0, len(height) - 1
        max_area = 0
        while low < high:
            max_area = max(max_area, 
                    (high - low) * min(height[low], height[high]))
            if height[low] < height[high]:
                low += 1
            else:
                high -= 1
        return max_area

File: LeetCode/Java/Solution/src/ios/ac/cn/ContainerWithMostWater.java
package ios.ac.cn;

public class ContainerWithMostWater {
	public int maxArea(int[] height) {
		if (height == null) {
			return 0;
		}
		int low = 0, high = height.length - 1;
		int max = 0;
		while (low < high) {
			if (height[low] < height[high]) {
				/*
				 * if height[low] < height[high], then the max area which is
				 * related to height[low] is (height[low] * (high - low))
				 */
				max = Math.max(max, height[low] * (high - low));
				low++;
			} else {
				max = Math.max(max, height[high] * (high - low));
				high--;
			}
		}
		return max;
	}
}
